flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/operations/JavaDataStreamQueryOperation.java

```diff
@@ -38,6 +38,11 @@
 @Internal
 public class JavaDataStreamQueryOperation<E> implements QueryOperation {
 
+	// JavaDataStreamQueryOperation represent a registered DataStream from `registerDataStream` method
+	// or a unregistered DataStream from `fromDataStream`. However the name under which the DataStream
+	// is registered is missing when converting a JavaDataStreamQueryOperation to a RelNode.
+	// So using `qualifiedName` to keep the original name.
+	private List<String> qualifiedName;
```

Comment:
Why not storing it in TableSchema?

Reply:
I don't fully understand. How to store this in `TableSchema`?

Reply:
I misunderstood this as field names..

Reply:
Can we use ObjectIdentifier here?